### ISSUE TITLE
[data grid] Scroll glitches after row reordering when a column is pinned right

### ISSUE THREAD

**Issue by ron-fm** [mui/mui-x#4300]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

When a column is pinned and drag and drop of columns initiated, the pinned header could move after some horizontal scrolling. Check the attached screen video

### Expected behavior 🤔

the pinned column shouldn't be affected on drag and drop

### Steps to reproduce 🕹

https://codesandbox.io/s/datagrid-v5-quick-start-forked-bg3t1h?file=/src/App.tsx
use sandbox above and following
Steps:

1. pin col3 to the right
2. drag and drop other header as you scroll and accidentally drop it in col3
3. scroll back and forth and if header still ok, drag and drop other column and repeat steps 2. I was able to re-create the issue just from dropping col4 to col3 while scrolling

https://user-images.githubusercontent.com/87013015/160511963-3f9c5e85-f41b-4053-8a20-06239a96a4fc.mov

### Your environment 🌎

<details>
  <summary>`npx @mui/envinfo`</summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Comment by DanailH** [comment]
Hey, @ron-fm thanks for raising this! I was able to reproduce the issue we'll have a look.

**Comment by m4theushw** [comment]
Will be fixed by #4280

Here's the CodeSandbox with the fix: https://codesandbox.io/s/datagrid-v5-quick-start-forked-cfje7u?file=/src/App.tsx
